### ISSUE TITLE
'uploader' information is null in YouTube playlist extractor

### ISSUE THREAD

**Issue by chrisr11** [blackjack4494/yt-dlc#173]
## Checklist

- [x] I've verified that I'm running youtube-dlc version **2020.11.11**
- [x] I've checked that all provided URLs are alive and playable in a browser
- [x] I've checked that all URLs and arguments with special characters are properly quoted or escaped
- [x] I've searched the bugtracker for similar bug reports including closed ones
- [x] I've read bugs section in FAQ

## Description

Hi, the uploader information is all null in the YouTube playlist extractor. More specifically, the 'uploader', 'uploader_url', and 'uploader_id'. Here's the response: https://ibb.co/hMz7HWG

This is the command that I'm using: youtube-dlc -v -4 -q --no-warnings -J --flat-playlist https://www.youtube.com/playlist?list=UU_aEa8K-EOJ3D6gOs7HcyNg

It used to include the uploader information before, but now it doesn't.

**Comment by stuart-little** [comment]
I don't know if this is in any way related, but as of today (Nov 11 2020) the latest `youtube-dl` (not `youtube-dlc`) doesn't even download playlists anymore (it sees 0 videos):

```
$ youtube-dl https://m.youtube.com/playlist?list=PL7yh-TELLS1HSmBcqYk9LhPYSWp31FWH2
[youtube:playlist] PL7yh-TELLS1HSmBcqYk9LhPYSWp31FWH2: Downloading webpage
[download] Downloading playlist: PL7yh-TELLS1HSmBcqYk9LhPYSWp31FWH2
[youtube:playlist] playlist PL7yh-TELLS1HSmBcqYk9LhPYSWp31FWH2: Downloading 0 videos
[download] Finished downloading playlist: PL7yh-TELLS1HSmBcqYk9LhPYSWp31FWH2
```

So I think Youtube may have changed the way they embed the requisite info in playlists.

**Comment by tygore587** [comment]
Also encounter this bug myself. Playlist title is set to NA all the time.

**Comment by thouu** [comment]
Having the same issue here, I'm trying to download videos in a folder that's the name of the uploader of the video, but instead of the channel name it just names the folder to "NA".

Here's my download script:
`youtube-dlc --sleep-interval 15 --download-archive "archive.log" -i --add-metadata --all-subs --embed-subs --embed-thumbnail --match-filter "playlist_title != 'Liked videos' & playlist_title != 'Favorites'" -f "(bestvideo[vcodec^=av01][height>=1080][fps>30]/bestvideo[vcodec=vp9.2][height>=1080][fps>30]/bestvideo[vcodec=vp9][height>=1080][fps>30]/bestvideo[vcodec^=av01][height>=1080]/bestvideo[vcodec=vp9.2][height>=1080]/bestvideo[vcodec=vp9][height>=1080]/bestvideo[height>=1080]/bestvideo[vcodec^=av01][height>=720][fps>30]/bestvideo[vcodec=vp9.2][height>=720][fps>30]/bestvideo[vcodec=vp9][height>=720][fps>30]/bestvideo[vcodec^=av01][height>=720]/bestvideo[vcodec=vp9.2][height>=720]/bestvideo[vcodec=vp9][height>=720]/bestvideo[height>=720]/bestvideo)+(bestaudio[acodec=opus]/bestaudio)/best" --merge-output-format mkv -o "$PWD/%(playlist_uploader)s/%(title)s - %(id)s.%(ext)s" "https://www.youtube.com/c/UnusAnnus/videos"

[Here are my verbose logs.](https://pastebin.com/ydJSukX6)

**Comment by numlockkey** [comment]
Also, the folder where the playlist is downloaded is not named properly. It names it with the same name as the actual URL.

So in this case the folder name is this:   PLDV1Zeh2NRsDGO4--qE8yH72HFL1Km93P

youtube-dlc -f 22/bestvideo+(140/bestaudio)/best -o "d:\YT Downloads\%%(uploader)s\%%(playlist_title)s\%%(playlist_index)s - %%(title)s.%%(ext)s" https://www.youtube.com/playlist?list=PLDV1Zeh2NRsDGO4--qE8yH72HFL1Km93P

**Comment by colethedj** [comment]
Can confirm I am having an issue with this too. It seems like the regex for extracting the playlist title, uploader, etc is no longer valid. It probably needs to be updated to use the json data. 

https://github.com/blackjack4494/yt-dlc/blob/d052b9a112fb7ae749a829dceba6e3289663a303/youtube_dlc/extractor/youtube.py#L2971-L2986

**Comment by BenarBloemendal** [comment]
any  progress in this issue ?

**Comment by colethedj** [comment]
I can see that #150 seems to have code in place that fixes this, but not sure myself what is happening with this.  For reference #151 was the one that got merged without this. Maybe we can merge the portion of the code in #150 that fixes this issue?

**Comment by Sinnoh** [comment]
Same Problem here when building from source (master-branch, d052b9a)
_%(playlist_title)s_ gets evaluated as NA in the output template.

I'd also suggest looking at #150 using their solution to fix this problem.

**Comment by pukkandan** [comment]
this should be fixed now - #245
